plugins/org.eclipse.fordiac.ide.gef/src/org/eclipse/fordiac/ide/gef/annotation/ResourceMarkerGraphicalAnnotationModel.java

```diff
@@ -137,6 +137,13 @@ public long getModificationStamp() {
 		return resource.getModificationStamp();
 	}
 
+	public final void setResource(final IResource resource) {
+		if (this.resource != resource) {
+			this.resource = resource;
+			reload();
```

Comment:
I considered this, but I am a bit worried that this might cause race conditions when there is a concurrent update from the `resourceChangeListener`, which is quite likely since this is also called when moving resources around. There may also be stale non-marker annotations left over afterwards, since a `reload()` intentionally only clears marker annotations.

Our textual relative `org.eclipse.ui.texteditor.ResourceMarkerAnnotationModel` also does not change the resource and instead uses a new annotation model for a when the resource changes.

However, I can also do that in a separate PR. Sorry for not getting around to this sooner.

Reply:
I mostly wanted to learn more about the annotation model. so no problem at all. I already expected taht my solution was to simple.